controller.go

```diff
@@ -286,35 +286,65 @@ type AllocateMachineArgs struct {
 	MinMemory int
 	Tags      []string
 	NotTags   []string
-	// Networks - list of networks (defined in MAAS) to which the machine must be
-	// attached. A network can be identified by the name assigned to it in MAAS;
-	// or by an ip: prefix followed by any IP address that falls within the
-	// network; or a vlan: prefix followed by a numeric VLAN tag, e.g. vlan:23
-	// for VLAN number 23. Valid VLAN tags must be in the range of 1 to 4094
-	// inclusive.
-	Networks    []string
-	NotNetworks []string
-	Zone        string
-	NotInZone   []string
-	AgentName   string
-	Comment     string
-	DryRun      bool
+	Zone      string
+	NotInZone []string
+	// Storage is an optional value listing one or more storage constraints.
+	// Format is an optional label, followed by an optional colon, then size
+	// (which is mandatory) followed by an optional comma seperated list of tags
+	// in parentheses.
+	//
+	//   Examples:
+	//
+	//   200(ssd,removable),400(ssd),300
+	//   - 200GB disk with ssd and removable tag
+	//   - 400GB disk with ssd tag
+	//   - 300GB disk
+	//
+	//  root:80(ssd),data:400
+	//  - 80+GB disk with ssd tag, name the constraint "root"
+	//  - 400+GB disk, name the consrtaint "data"
+	Storage string
```

Comment:
Wouldn't it be cleaner to accept this as a []string or better yet a []StorageSpec, where StorageSpec has fields for the size, labels, and tags. The caller shouldn't have to care about how MAAS wants this to be formatted.